jmx-metrics/src/integrationTest/java/io/opentelemetry/contrib/jmxmetrics/AbstractIntegrationTest.java

```diff
@@ -0,0 +1,233 @@
+
+    List<String> scraperCommand = new ArrayList<>();
+    scraperCommand.add("java");
+    scraperCommand.add("-cp");
+    scraperCommand.add("/app/OpenTelemetryJava.jar");
+    scraperCommand.add("-Dotel.jmx.username=cassandra");
+    scraperCommand.add("-Dotel.jmx.password=cassandra");
+    scraperCommand.add(
+        "-Dotel.exporter.otlp.endpoint=http://host.testcontainers.internal:"
+            + otlpServer.httpPort());
+    scraperCommand.add("io.opentelemetry.contrib.jmxmetrics.JmxMetrics");
+    scraperCommand.add("-config");
+
+    if (configFromStdin) {
+      String cmd = String.join(" ", scraperCommand);
+      scraperCommand = Arrays.asList("sh", "-c", "cat /app/" + configName + " | " + cmd + " -");
+    } else {
+      scraperCommand.add("/app/" + configName);
+    }
+
+    scraper =
+        new GenericContainer<>("openjdk:8u272-jre-slim")
+            .withNetwork(Network.SHARED)
+            .withCopyFileToContainer(
+                MountableFile.forHostPath(scraperJarPath), "/app/OpenTelemetryJava.jar")
+            .withCopyFileToContainer(
+                MountableFile.forClasspathResource("script.groovy"), "/app/script.groovy")
+            .withCopyFileToContainer(
+                MountableFile.forClasspathResource(configName), "/app/" + configName)
+            .withCommand(scraperCommand.toArray(new String[0]))
+            .withStartupTimeout(Duration.ofSeconds(120))
+            .waitingFor(Wait.forLogMessage(".*Started GroovyRunner.*", 1));
+    scraper.start();
+  }
+
+  @AfterAll
+  void afterAll() {
+    otlpServer.stop();
+  }
+
+  @BeforeEach
+  void beforeEach() {
+    otlpServer.reset();
+  }
+
+  protected void waitAndAssertMetrics(Consumer<Metric>... assertions) {
+    await()
+        .atMost(Duration.ofSeconds(30))
+        .untilAsserted(
+            () -> {
+              List<Metric> metrics =
+                  otlpServer.getMetrics().stream()
+                      .map(ExportMetricsServiceRequest::getResourceMetricsList)
+                      .flatMap(
+                          rm ->
+                              rm.stream()
+                                  .map(ResourceMetrics::getInstrumentationLibraryMetricsList))
+                      .flatMap(Collection::stream)
+                      .filter(
+                          ilm ->
+                              ilm.getInstrumentationLibrary()
+                                  .getName()
+                                  .equals("io.opentelemetry.contrib.jmxmetrics"))
+                      .flatMap(ilm -> ilm.getMetricsList().stream())
+                      .collect(Collectors.toList());
+
+              assertThat(metrics).isNotEmpty();
+
+              for (Consumer<Metric> assertion : assertions) {
+                assertThat(metrics).anySatisfy(assertion);
+              }
+            });
+  }
+
+  protected void assertGauge(Metric metric, String name, String description) {
+    assertGauge(metric, name, description, "µs");
```

Comment:
This could be a little unintuitive.  Isn't unitless `assertGauge` used more than microseconds?

Reply:
Thanks for the callout - indeed I added these only based on the first test I migrated. In the end, it seems there is no clear default so leaving it out is best